File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.py
def solution(keyinput, board):

    max_x = board[0] // 2
    max_y = board[1] // 2
    x = 0
    y = 0
    
    for key in keyinput:
        if key == "left":
            if x == max_x * -1: pass
            else: x -= 1
        elif key == "right":
            if x == max_x: pass
            else: x += 1
        elif key == "up":
            if y == max_y: pass
            else: y += 1
        elif key == "down":
            if y == max_y * -1: pass
            else: y -= 1
            
    return x, y

File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.java
class Solution {
    public int[] solution(String[] keyinput, int[] board) {
        int[] answer = {0, 0};
        int xMax = board[0] / 2;
        int yMax = board[1] / 2;
        
        for (String move: keyinput) {
            if (move.equals("left")) {
                if (answer[0] == xMax * -1) {
                    continue;
                } else {
                    answer[0] -= 1;
                }
            } else if (move.equals("right")) {
                if (answer[0] == xMax) {
                    continue;
                } else {
                    answer[0] += 1;
                }
            } else if (move.equals("up")) {
                if (answer[1] == yMax) {
                    continue;
                } else {
                    answer[1] += 1;
                }
            } else if (move.equals("down")) {
                if (answer[1] == yMax * -1) {
                    continue;
                } else {
                    answer[1] -= 1;
                }
            }
        }
        
        return answer;
    }
}